core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -1627,6 +1627,16 @@ object ZIO extends ZIO_R_Any {
       }
   }
 
+  class AccessPartiallyApplied[R >: LowerR] {
+    def apply[A](f: R => A): ZIO[R, Nothing, A] =
+      accessM(f.andThen(succeed))
```

Comment:
This will be slightly slower than doing it the 'dumb' way because `succeed` must be wrapped in its own lambda, then a composite one generated by `andThen`.

Reply:
Right. Aggressively inline everything!